src/main/java/org/javarosa/xpath/expr/XPathSortFunc.java

```diff
@@ -52,6 +53,9 @@ protected Object evalBody(DataInstance model, EvaluationContext evalContext, Obj
             sortedList = sortSingleList(FunctionUtils.toString(evaluatedArgs[0]),
                     FunctionUtils.toBoolean(evaluatedArgs[1]));
         }
+        if (sortedList.size() == 0) {
```

Comment:
Wondering why can't we just return the same empty list instead of throwing an error ?

Reply:
Good question, I hadn't thought about that. Given that we were crashing previously when this was called on an empty list I'm inclined to not change the behavior in this PR. I also think its pretty likely that if `sort()` is called with an empty list then that indicates a user error and we're better off crashing rather than hiding the issue.

Reply:
Hmm.. Though calling sort with a Xpath reference to a list which can sometimes evaluate to a empty list would also result in this error and won't always signify an error on user's part unless we always want that user to check the size of the list represented by xpath ref before calling sort with it.

Reply:
Hey yea I definitely agree, good call Shubham-- it's completely plausible that an app builder would want to call `sort()` on something that will sometimes evaluate to empty and and sometimes not. Returning the empty list also seems like totally correct behavior to me from a theoretical standpoint.

Reply:
I think being that strict about arguments is in line with how other our other XPath functions behave and we've had the onus be on users to account for null-conditions in their apps using `coalesce` when desired

Reply:
I agree with the Null-conditions part though don't think we should treat an empty list same as null. Like I would be pretty sad if Java Collections Sort throw an error for a var list that can be empty sometimes and I would have to check both for not null and not empty before calling sort.  

Though since we are already crashing for an empty list currently, I think throwing this error makes sense in context of this PR.